assemble/bin/accumulo-cluster

```diff
@@ -43,52 +43,74 @@ function invalid_args {
   exit 1
 }
 
-function verify_config {
+function parse_config {
   if [[ -f ${conf}/slaves ]]; then
     echo "ERROR: A 'slaves' file was found in ${conf}/"
-    echo "Accumulo now reads tablet server hosts from 'tservers' and requires that the 'slaves' file not be present to reduce confusion."
+    echo "Accumulo now uses cluster host configuration information from 'cluster.yml' and requires that the 'slaves' file not be present to reduce confusion."
     echo "Please rename the 'slaves' file to 'tservers' or remove it if both exist."
     exit 1
   fi
 
-  if [[ ! -f ${conf}/tservers ]]; then
-    echo "ERROR: A 'tservers' file was not found at ${conf}/tservers"
-    echo "Please make sure it exists and is configured with tablet server hosts."
+  if [[ ! -f ${conf}/cluster.yml ]]; then
+    echo "ERROR: A 'cluster.yml' file was not found at ${conf}/cluster.yml"
+    echo "Please make sure it exists and is configured with the host information. Run 'create-config' to create an example configuration."
     exit 1
   fi
 
-  unset manager1
-  if [[ -f "${conf}/$manager_file" ]]; then
-    manager1=$(grep -Ev '(^#|^\s*$)' "${conf}/$manager_file" | head -1)
+  config=$(${accumulo_cmd} org.apache.accumulo.core.conf.cluster.ClusterConfigParser ${conf}/cluster.yml)
+
+  declare -A HOSTS
+
+  HOSTS["MANAGERS"]=$(grep managers $config | awk -F':' '{ print $2 }')
+  HOSTS["MONITORS"]=$(grep monitors $config | awk -F':' '{ print $2 }')
+  HOSTS["GC"]=$(grep gc $config | awk -F':' '{ print $2 }')
+  HOSTS["TRACER"]=$(grep tracer $config | awk -F':' '{ print $2 }')
+  HOSTS["TSERVERS"]=$(grep tservers $config | awk -F':' '{ print $2 }')
+
+  grep "compaction" $config $2>1 > /dev/null
+  compaction_in_config=$?
+
+  if [[ $compaction_in_config -eq 0 ]]; then
+    HOSTS["COMPACTION_COORDINATORS"]=$(grep "compaction.coordinators" $file | awk -F':' '{ print $2 }')
+    HOSTS["COMPACTION_QUEUES"]=$(grep "compaction.compactors.queues" $file | awk -F':' '{ print $2 }')
+
+    IFS_ORIG=$IFS
+    IFS=","
+    for queue in ${HOSTS["COMPACTION_QUEUES"]}; do
+      HOSTS["COMPACTORS_$queue"]=$(grep "compaction.compactors.$queue" $file | awk -F':' '{ print $2 }')
```

Comment:
could do `eval COMPACTORS_$queue=...` instead of an associative array for these. Or, put the queue name at the front, in a regular bash array, as in:

```bash
COMPACTOR_QUEUES=($queue:'compactors,here')
```
You can use += on bash 3.2 on Macs, I've read. But, another way to append to an array is:
```bash
COMPACTOR_QUEUES=("${COMPACTOR_QUEUES[@]}" $queue:'compactors,here')
```